main/inc/lib/webservices/Rest.php

```diff
@@ -1646,4 +1647,29 @@ public function createSessionFromModel($modelSessionId, $sessionName, $startDate
 
         return [$newSessionId];
     }
+
+    /**
+     * @param $sessionId
+     * @param $loginName
+     * @return array
+     * @throws Exception
+     */
+    public function subscribeUserToSessionFromUsername($sessionId, $loginName)
+    {
+        if (!SessionManager::isValidId($sessionId)) {
+            throw new Exception(get_lang('SessionNotFound'));
+        }
+
+        $userId = UserManager::get_user_id_from_username($loginName);
+        if (False === $userId) {
+            throw new Exception(get_lang('UserNotFound'));
+        }
+
+        $subscribed = SessionManager::subscribeUsersToSession($sessionId, [$userId]);
+        if (!$subscribed) {
+            throw new Exception(get_lang('UserNotSubscribed'));
+        }
+
+        return [true];
```

Comment:
It would be preferible to use a simple integer, except if required as array elsewhere.